ow_lander/scripts/dig_circular_action_client.py

```diff
@@ -4,45 +4,73 @@
 # Research and Simulation can be found in README.md in the root directory of
 # this repository.
 
+# Default trenching values
+# goal.x_start = 1.65
+# goal.y_start = 0
+# goal.depth = 0.01
+# goal.parallel = False
+# goal.ground_position = constants.DEFAULT_GROUND_HEIGHT
+
+# General trenching values
+# goal.x_start = 1.55
+# goal.y_start = 0.2
+# goal.depth = 0.045
+# goal.parallel = False
+# goal.ground_position = -0.155
+
 import rospy
 import actionlib
 import ow_lander.msg
 import constants
+import argparse
+
 
 def DigCircular_client():
- 
-    client = actionlib.SimpleActionClient('DigCircular', ow_lander.msg.DigCircularAction)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('x_start', type=float,
+                        help='X coordinate of trenching starting point', nargs='?', default=1.65, const=0)
+    parser.add_argument('y_start', type=float,
+                        help='Y coordinate of trenching starting point', nargs='?', default=0.0, const=0)
+    parser.add_argument('depth', type=float,
+                        help='Desired depth', nargs='?', default=0.01, const=0)
+    parser.add_argument('parallel', type=bool,
+                        help='If True, resulting trench is parallel to arm. If False, perpendicular to arm', nargs='?', default=0, const=0)
```

Comment:
`bool` doesn't behave how one would expect here. From the [argparse docs](https://docs.python.org/3/library/argparse.html#type)
> The bool() function is not recommended as a type converter. All it does is convert empty strings to False and non-empty strings to True. This is usually not what is desired.

Meaning when I type "false" for the parallel argument, I actually wind up with a value of **True**.
```
$ python dig_circular_action_client.py 1.65 0 0.01 false
[INFO] [1636074722.383847, 0.000000]: Requested x_start: 1.65
[INFO] [1636074722.386141, 0.000000]: Requested y_start: 0.0
[INFO] [1636074722.388379, 0.000000]: Requested depth: 0.01
[INFO] [1636074722.390086, 0.000000]: Requested parallel: True
[INFO] [1636074722.391243, 0.000000]: Requested ground_position: -0.155
[INFO] [1636074868.966462, 1916530139.065000]: Result: final: 
  x: 1.64094870742
  y: -0.00556842319074
  z: -7.24609931862
```

There are *a lot* of different ways to handle this. Here's the one I settled on because it's short and does not require the user to be overly careful about their capitalization.
```suggestion
    parser.add_argument('parallel', type=lambda x: eval(x[0].upper() + x[1:].lower()),
                        help='If True, resulting trench is parallel to arm. If False, perpendicular to arm', nargs='?', default=False)
```
Alternatively, if you already have a module of common functions, you could make it a function and just pass that.
```suggestion
    parser.add_argument('parallel', type=str2bool,
                        help='If True, resulting trench is parallel to arm. If False, perpendicular to arm', nargs='?', default=False)
```
With the function defined elsewhere the same way as the lambda was
```Python
def str2bool(val):
    return eval(val[0].upper() + x[1:].lower())
```
I also changed the default value because writing out **False** is much more clear than **0**.

Reply:
Thank you for this.. I have included the suggestion.